docs/architecture/README.md

```diff
@@ -1,5 +1,8 @@
 ---
-position: 4
+sections:
+  - application
+  - layer
+  - feature
```

Comment:
section and page naming is a bit confusing. let's not use section for those inner pages, let's call them pages. now section only refers to main pages in the top menu like `blueprints`, `features` etc.

so this way, 

- this array should be renamed as `pages`
- side menu items should be called as `PAGES` instead of `SECTIONS`
  - `ON THIS PAGE` would be more consistent now
- rename anything related to a page as page instead of section
- rename anything related to a section as section instead of anything else